src/scripts/clipperUI/componentBase.ts

```diff
@@ -66,6 +71,60 @@ export abstract class ComponentBase<TState, TProps> {
 		};
 	}
 
+	/*
+	 * Helper which handles tabIndex, clicks, and keyboard navigation for a component that is part of an Aria Set
+	*
+	 * Also hides the outline if they are using a mouse, but shows it if they are using the keyboard
+	 * (idea from http://www.paciellogroup.com/blog/2012/04/how-to-remove-css-outlines-in-an-accessible-manner/)
+	 */
+	enableAriaInvoke({callback, tabIndex, args, idOverride, ariaSetName, ariaSetDirection}: EnableAriaParams) {
+		if (callback) {
+			callback = callback.bind(this, args);
+		}
+
+		let invokeAttributes = this.enableInvoke({callback: callback, tabIndex: tabIndex, args: args, idOverride: idOverride});
+		let oldKeyUp = invokeAttributes.onkeyup;
+
+		invokeAttributes.onkeyup = (e: KeyboardEvent) => {
+			e.preventDefault();
+			let currentTargetElement = e.currentTarget as HTMLElement;
+
+			oldKeyUp(e);
+
+			if (e.which === Constants.KeyCodes.home) {
+				let firstInSet = 1;
+				ComponentBase.focusOnButton(ariaSetName, firstInSet);
+			} else if (e.which === Constants.KeyCodes.end) {
+				let lastInSet = parseInt(currentTargetElement.getAttribute("aria-setsize"), 10);
+				ComponentBase.focusOnButton(ariaSetName, lastInSet);
+			}
+
+			let posInSet = parseInt(currentTargetElement.getAttribute("aria-posinset"), 10);
+			let increaseButton = ariaSetDirection === AriaNavDirection.Vertical ? Constants.KeyCodes.up : Constants.KeyCodes.left;
```

Comment:
Shouldn't the variable names be the other way around? When using up/left, we're decreasing the index. When using down/right, we're increasing the index